server/src/test/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregatorTests.java

```diff
@@ -13,20 +13,43 @@
 import org.apache.lucene.index.DirectoryReader;
 import org.apache.lucene.index.IndexReader;
 import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
 import org.apache.lucene.store.Directory;
 import org.apache.lucene.tests.index.RandomIndexWriter;
+import org.elasticsearch.core.Releasables;
+import org.elasticsearch.index.mapper.MappedFieldType;
 import org.elasticsearch.index.mapper.NumberFieldMapper;
 import org.elasticsearch.search.aggregations.AggregationExecutionContext;
 import org.elasticsearch.search.aggregations.AggregatorFactories;
 import org.elasticsearch.search.aggregations.AggregatorTestCase;
 import org.elasticsearch.search.aggregations.InternalAggregation;
 import org.elasticsearch.search.aggregations.LeafBucketCollector;
 import org.elasticsearch.search.aggregations.support.AggregationContext;
+import org.junit.After;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 
 public class BucketsAggregatorTests extends AggregatorTestCase {
 
+    private List<AggregationContext> toRelease = new ArrayList<>();
+
+    @Override
+    protected AggregationContext createAggregationContext(IndexSearcher indexSearcher, Query query, MappedFieldType... fieldTypes)
+        throws IOException {
+        AggregationContext context = super.createAggregationContext(indexSearcher, query, fieldTypes);
+        // Generally, we should avoid doing this, but
```

Comment:
"but this test doesn't test things to do with reduction so it should be safe here"

Reply:
I just sort of trailed off mid sentence there.  Not sure what distracted me.

Reply:
happens